src/smartpeak/source/core/SessionLoaderGenerator.cpp

```diff
@@ -0,0 +1,100 @@
+// --------------------------------------------------------------------------
+//   SmartPeak -- Fast and Accurate CE-, GC- and LC-MS(/MS) Data Processing
+// --------------------------------------------------------------------------
+// Copyright The SmartPeak Team -- Novo Nordisk Foundation 
+// Center for Biosustainability, Technical University of Denmark 2018-2021.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Douglas McCloskey, Bertrand Boudaud $
+// $Authors: Douglas McCloskey, Bertrand Boudaud $
+// --------------------------------------------------------------------------
+
+#include <SmartPeak/core/SessionLoaderGenerator.h>
+#include <plog/Log.h>
+
+namespace SmartPeak
+{
+
+  void SessionLoaderGenerator::onApplicationProcessorStart(const std::vector<std::string>& commands)
+  {
+    for (const auto& command : commands)
+    {
+      if (command == "CLEAR_DATA")
+      {
+        loading_workflow_commands_.clear();
+      }
+      else if (command == "LOAD_RAW_DATA")
+      {
+        loading_workflow_commands_.emplace(0, "LOAD_RAW_DATA");
+      }
+      else if (command == "STORE_FEATURES")
+      {
+        loading_workflow_commands_.emplace(10, "LOAD_FEATURES");
+      }
+      else if (command == "MAP_CHROMATOGRAMS")
+      {
+        loading_workflow_commands_.emplace(20, "MAP_CHROMATOGRAMS");
+      }
+      else if (command == "EXTRACT_CHROMATOGRAM_WINDOWS")
+      {
+        loading_workflow_commands_.emplace(21, "EXTRACT_CHROMATOGRAM_WINDOWS");
+      }
+      else if (command == "ZERO_CHROMATOGRAM_BASELINE")
+      {
+        loading_workflow_commands_.emplace(22, "ZERO_CHROMATOGRAM_BASELINE");
+      }
```

Comment:
"MERGE_SPECTRA" and "EXTRACT_SPECTRA_WINDOWS" from the FIA-MS workflow could also be added.  Their inverse commands are the same.